src/sentry/workflow_engine/handlers/condition/event_frequency_handlers.py

```diff
@@ -0,0 +1,245 @@
+import contextlib
+from abc import ABC, abstractmethod
+from collections import defaultdict
+from collections.abc import Callable, Mapping
+from datetime import datetime, timedelta
+from typing import Any, Literal, TypedDict
+
+from django.db.models import QuerySet
+
+from sentry import tsdb
+from sentry.issues.constants import get_issue_tsdb_group_model
+from sentry.issues.grouptype import GroupCategory, get_group_type_by_type_id
+from sentry.models.group import Group
+from sentry.tsdb.base import TSDBModel
+from sentry.utils import json
+from sentry.utils.iterators import chunked
+from sentry.utils.snuba import options_override
+from sentry.workflow_engine.models.data_condition import Condition
+from sentry.workflow_engine.registry import condition_handler_registry
+from sentry.workflow_engine.types import DataConditionHandler, DataConditionResult
+
+SNUBA_LIMIT = 10000
+STANDARD_INTERVALS: dict[str, tuple[str, timedelta]] = {
+    "1m": ("one minute", timedelta(minutes=1)),
+    "5m": ("5 minutes", timedelta(minutes=5)),
+    "15m": ("15 minutes", timedelta(minutes=15)),
+    "1h": ("one hour", timedelta(hours=1)),
+    "1d": ("one day", timedelta(hours=24)),
+    "1w": ("one week", timedelta(days=7)),
+    "30d": ("30 days", timedelta(days=30)),
+}
+COMPARISON_INTERVALS: dict[str, tuple[str, timedelta]] = {
+    "5m": ("5 minutes", timedelta(minutes=5)),
+    "15m": ("15 minutes", timedelta(minutes=15)),
+    "1h": ("one hour", timedelta(hours=1)),
+    "1d": ("one day", timedelta(hours=24)),
+    "1w": ("one week", timedelta(days=7)),
+    "30d": ("30 days", timedelta(days=30)),
+}
+
+
+class _QSTypedDict(TypedDict):
+    id: int
+    type: int
+    project_id: int
+    project__organization_id: int
+
+
+class BaseEventFrequencyConditionHandler(DataConditionHandler[int], ABC):
```

Comment:
the value is an `int`, because slow conditions are not evaluated in the general `process_workflows` function. after we fetch the single value in delayed processing we can compare it with the threshold. for percent comparison this will need to become an array (to compare percent change)